64.md

```diff
@@ -0,0 +1,55 @@
+NIP-64
+======
+
+Inbox model
+-----------
+
+`draft` `optional`
+
+Suppose **Sarah** wants to subscribe to the notes **Walter** writes.
+
+She must let Walter know she is following him. The first step is to find Walter's _inbox list_:
+
+```jsonc
+{
+  "kind": 10064,
+  "pubkey": "<walter>",
+  "tags": [
+    ["relay", "wss://walter.inbox"]
+  ]
+}
+```
+
+Now Sarah can publish a `kind:6401` _follow intent_ to `wss://walter.inbox` tagging Walter:
+
+```jsonc
+{
+  "kind": 6401,
+  "pubkey": "<sarah>"
+  "tags": [
+    ["p", "<walter>"],
+    ["relay", "wss://sarah.inbox"],
+    ["relay", "wss://other.inbox"]
+  ]
+}
+```
+
+Sarah also publishes these to her own inbox relays (`wss://sarah.inbox` and `wss://other.inbox`) so they are aware.
+
+From the Walter side, whenever he is about to publish a note, he downloads all the `kind:6401` events from his inbox relays and sends his note to all the relays he finds in these.
+
+Whenever Sarah's inbox relay receive a note from Walter, they will know that Sarah is following him, so they index that note in a way that it is associated with Sarah. If the relays receive a note from someone else they can simply reject the note.
+
+When Sarah turns on her client she will connect to her inbox relay, perform [NIP-42](42.md) `AUTH` and create a subscription that doesn't specify `"authors"` (for example, `["REQ", "_", {}]`). The relays, aware of who she is and who she is following, can deliver to her just Walter's notes.
```

Comment:
This breaks how absent `authors` filter works.

Instead of NIP-42, better to filter like this `{ "nip64": "<sarah-pubkey>", kinds: [1], ...etc }`. The `nip64` key clearly asks for relay to change how an absent `authors` filter should be treated.

It is ok to let anyone see the "feed" as Sarah cause follow list is public.

Reply:
This introduces unnecessary complexity. A new term to remember, libraries would have to add support for this and everything would be awful.

I don't see any issue with using an empty `"authors"` and expecting the relay to treat that in a certain way, as long as the relay is carefully chosen to be treated differently by the client. A relay is already allowed to return anything they want _as long as it matches the filter_, it isn't mandated or expected to _always return everything_.

There are already other experiments out there of relay that behave differently when `AUTH`ed and when not and I hope to see more.

Also specifying the `AUTH` stuff from the beginning is nice as it allows for private feeds in cool ways that, although not explicitly specified in this NIP, could be implemented somewhat easily.

Reply:
Immediately after writing the comment above I read this note from @mikedilger: https://njump.me/nevent1qy88wumn8ghj7mn0wvhxcmmv9uqjqamnwvaz7tmwdaehgu3wv45kuatwv3a8wctw0f5kwtnnwpskxef0qyfhwumn8ghj7mmxve3ksctfdch8qatz9uqzpv48826guadrdezm5eyswj6m2vjgq9pewzz309v427jcv33yrgqdlvlk6v

Reply:
I must have missed this entire paragraph. You are suggesting to put the smarts in the relay.  I thought Walter would just tag Sarah when he puts her notes into her inbox relay.  If the relay really handles the subscriptions as a service without the tagging, then of course this is not synonymous with a NIP-65 inbox relay and it should be specified differently.

Reply:
I don't think there should be any tagging. Walter writes one note, signs it once and sends it to all the relays of people who have previously announced their intent to follow him (and also sends it to his own outbox relays). If he had to write a different note for each I think this would be a cumbersome nightmare.